### PR TITLE
State: align ocispec-rs with spec-go and introduce State

### LINKED ISSUE #169: oci-spec-rs/Runtime needs add support State
Currently, the oci-spec-rs/Runtime doesn't support State yet. But in some other opensource projects needs such item. And it needs align oci-spec-rs with [OCI Runtime-Spec](https://github.com/opencontainers/runtime-spec/blob/main/specs-go/state.go)

### PR DESCRIPTION
**Pull request by Apokleos**
Fixes #169

### CODE REVIEW

**Comment by Apokleos** [comment]
Hi @utam0k @saschagrunert PTAL!

File: src/runtime/state.rs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    /// Created indicates that the runtime has finished the create operation,
+    /// but the container exists but has not been run yet.
+    Created,
+
+    /// Running indicates that the container process has executed the
+    /// user-specified program but has not exited
+    Running,
+
+    /// Paused indicates that the container exists, but all processes are paused.
+    Paused,
```
Comment (saschagrunert) [inline src/runtime/state.rs:24]: Is this something official? I can't see it in https://github.com/opencontainers/runtime-spec/blob/main/specs-go/state.go

File: src/runtime/state.rs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+    /// StateStopped indicates that the container process has exited,
+    /// and does not have a created or running process.
+    Stopped,
+
+    /// Unknown
+    #[default]
+    Unknown,
```
Comment (saschagrunert) [inline src/runtime/state.rs:32]: Same for this, do we need that?

File: src/runtime/state.rs [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    /// descriptor in the `SCM_RIGHTS` array.
+    #[serde(default)]
+    fds: Vec<String>,
+
+    /// pid is the process ID as seen by the runtime.
+    #[serde(default)]
+    pid: i32,
+
+    /// opaque metadata.
+    #[serde(default)]
```
Comment (saschagrunert) [inline src/runtime/state.rs:138]:
```suggestion
    #[serde(default, skip_serializing_if = "Option::is_none")]
```

File: src/runtime/state.rs (hunk 1 above)
Comment (Apokleos) [reply src/runtime/state.rs:24]: I'm so sorry @saschagrunert
I think I misunderstood the State of things and mixed them up. I need to rethink it. Thank you for your reminder.

**Comment by Apokleos** [comment]
Sorry @saschagrunert @utam0k for that my misunderstanding of this state, I think I should close it.  Thx!

**Pull request closed without merging**